### DISCUSSION TITLE
Possible issue between Adafruit TinyUSB MSC and Timerinterrupt libraries.

### DISCUSSION THREAD

**Discussion by bluegrassrules** [khoih-prog/TimerInterrupt#25] [Q&A]
Not sure if this is the right place to ask my questions – start a discussion.  First off, your timerinterrupt library is fantastic!  Thanks for sharing it with everyone.  Second, a little background.  I’m using your library on an Adafruit nRF52840 Feather Express along with an Adalogger RTC with SD card.  The SD card is used by the Adafruit to read and write files from external EEPROMs. Then Adafruit TinyUSB’s library (v1.7.1) is used to have the SD card act as a mass storage device on a computer.  That is, the SD card appears as an USB flash drive on the computer.  I’m using your library (v1.7.0) to blink and flicker 6 LEDs in the system.  I have this working perfectly on 16 systems.  My problem is I have one, number 17, that is working except for having a problem where the interrupt timers used by the LEDs don’t work when the SD card is mounted on the computer.  System 17 has the exact same hardware and software as the other working 16.  They all have the same bootloader, v0.62.  I’ve checked and the interrupt service routine that toggles the LED’s digital pin is not being called until I unmount the flash drive or I never initialize the MSC usb_msc.begin().  If I call ISR_Timer.isEnabled() for the timer that should be blinking the LED, it returns true.

I guess my question is, do you know of any issues between your library and the Adafruit TinyUSB mass storage class?  Any suggestions on how to debug your library to try and determine what the issue is for this one specific system?

Thanks,

**Comment by khoih-prog** [comment]
Without the correct info, such as
- your board in nRF52840 ??  But this `TimerInterrupt` is for AVR Mega, Nano, UNO, 32U4, etc. Did you post in the wrong library (to be [**TimerInterrupt_Generic**](https://github.com/khoih-prog/TimerInterrupt_Generic) ??)
- Your [**MRE**](https://stackoverflow.com/help/minimal-reproducible-example) so that anyone can duplicate the issue

I think I only can use my cloudy crystal ball to guesstimate that it's possibly there is some interrupt conflict and you can try to change the interrupt and test.

For example, if nRF52840, change from `NRF_TIMER_1` to `NRF_TIMER_2`-`NRF_TIMER_4`

```
// Depending on the board, you can select NRF52 Hardware Timer from NRF_TIMER_1-NRF_TIMER_4 (1 to 4)
// If you select the already-used NRF_TIMER_0, it'll be auto modified to use NRF_TIMER_1

// Init NRF52 timer NRF_TIMER1
NRF52Timer ITimer(NRF_TIMER_1);
```

I'm not aware of any conflict between TimerInterrupt_Generic library and Adafruit TinyUSB library. Anyway, I don't have `Adalogger RTC with SD card` to try and see if there is any `hardware` conflict besides other conflicts.

If you still can not solve the issue yourself, I suggest you post the issue on `Adafruit Forum` or `Adafruit TinyUSB library` for help.

Good Luck,

**Reply by bluegrassrules** [reply]
Thanks for your response.  Yes, I miss posted.  I'm using the [NRF52_TimerInterrupt](https://github.com/khoih-prog/NRF52_TimerInterrupt).  Should I switch to Timerinterrupt_Generic? Or should either work?

I'll try to put together a MRE, if your suggestion doesn't work.

I'm using NRF_TIMER_3.  I'll try another one.

Thanks for the suggestion of the other forums.
Thanks,

**Reply by bluegrassrules** [reply]
Just an update on status.  I tried all 4 hardware timers - no change.  Posted questions to Adafruit forum.
Thanks,
